zenoh/src/session.rs

```diff
@@ -499,135 +562,49 @@ impl Session {
     pub fn config(&self) -> &Notifier<Config> {
         self.runtime.config()
     }
+}
```

Comment:
This is actually a sneaky breaking change: the `SessionDeclarations` trait will have to be in scope for the methods to be called on `Session`, whereas this is currently unnecessary.

While I agree this makes a lot of sense, we should probably have a quick internal chat about whether or not we want to make that break (we have other more localized breaking changes in the next release, so I'm open to it)